docs(hub) update rate-limiting docs (#2139)
Kong/docs.konghq.com: app/_hub/kong-inc/rate-limiting/index.md

publisher: Kong Inc.
redirect_from:
  - /enterprise/0.35-x/rate-limiting/
version: 2.2.0

desc: Rate-limit how many HTTP requests a developer can make
description: |

>   Rate limit how many HTTP requests a developer can make in a given period of seconds, minutes, hours, days, months, or years. If the underlying Service/Route (or deprecated API entity) has no authentication layer, the **Client IP** address will be used, otherwise the Consumer will be used if an authentication plugin has been configured.

Reviewer: unsure: requests a developer can make? maybe client application, or "...requests can be made..." (yeah, passive, but less specific about request origin)

praise: I like the Oxford comma!  (twice in the same paragraph!)

Author: Good point, rephrased it

    - name: second
      required: semi
      value_in_examples: 5
      description: The number of HTTP requests the developer can make per second. At least one limit must exist.
    - name: minute
      required: semi
      description: The number of HTTP requests the developer can make per minute. At least one limit must exist.
    - name: hour
      required: semi
      value_in_examples: 10000
      description: The number of HTTP requests the developer can make per hour. At least one limit must exist.
    - name: day
      required: semi
      description: The number of HTTP requests the developer can make per day. At least one limit must exist.
    - name: month
      required: semi
      description: The number of HTTP requests the developer can make per month. At least one limit must exist.
    - name: year
      required: semi

>       description: The number of HTTP requests the developer can make per year. At least one limit must exist.

Reviewer: i guess the "at least one limit must exist" refers to the whole set.  maybe it could be moved after these params list, and then enumerate them?  like "at least one of second, minute, hour, day, month, or year must specify a limit"

Author: Added an observation at the end.

      required: false
      default: '`consumer`'
      description: |

>         The entity that will be used when aggregating the limits: `consumer`, `credential`, `ip`, `service`, `header`. If the `consumer`, the `credential` or the `service` cannot be determined, the system will always fallback to `ip`. If value `header` is chosen, the `header_name` configuration has to be provided.

Reviewer 2: ```suggestion
        The entity that will be used when aggregating the limits: `consumer`, `credential`, `ip`, `service`, or `header`. If the `consumer`, the `credential`, or the `service` cannot be determined, the system will always fall back to `ip`. If the value `header` is chosen, the `header_name` parameter must be configured.
```

    - name: header_name
      required: semi

>       description: Header name to be used if `limit_by` is set to `header`.

Reviewer 2: ```suggestion
      description: The header name to aggregate by if `limit_by` is set to `header`.
```
